paper-toast.html

```diff
@@ -123,41 +124,60 @@
      * @method show
      */
     show: function() {
-      if (PaperToast.currentToast) {
-        PaperToast.currentToast.hide();
-      }
-      PaperToast.currentToast = this;
-      this.removeAttribute('aria-hidden');
-      this._setVisible(true);
-      this.fire('iron-announce', {
-        text: this.text
-      });
-      this.debounce('hide', this.hide, this.duration);
+      this.open();
     },
 
     /**
      * Hide the toast
      */
     hide: function() {
-      this.setAttribute('aria-hidden', 'true');
-      this._setVisible(false);
+      this.close();
     },
 
     /**
-     * Toggle the opened state of the toast.
-     * @method toggle
+     * Overridden from `IronOverlayBehavior`.
+     * Called when the value of `opened` changes.
      */
-    toggle: function() {
-      if (!this.visible) {
-        this.show();
-      } else {
-        this.hide();
+    _openedChanged: function() {
+      if (this.opened) {
+        if (PaperToast.currentToast && PaperToast.currentToast !== this) {
+          PaperToast.currentToast.close();
+        }
+        PaperToast.currentToast = this;
+        this.fire('iron-announce', {
+          text: this.text
+        });
+        // auto-close if duration is a positive finite number
+        if (this.duration > 0 && this.duration !== Infinity) {
+          this.debounce('close', this.close, this.duration);
+        }
+      } else if (PaperToast.currentToast === this) {
+        PaperToast.currentToast = null;
       }
+      Polymer.IronOverlayBehaviorImpl._openedChanged.apply(this, arguments);
     },
 
-    _visibleChanged: function(visible) {
-      this.toggleClass('paper-toast-open', visible);
+    /**
+     * Overridden from `IronOverlayBehavior`.
+     */
+    _renderOpened: function() {
+      this.classList.add('paper-toast-open');
+    },
+    /**
+     * Overridden from `IronOverlayBehavior`.
+     */
+    _renderClosed: function() {
+      this.classList.remove('paper-toast-open');
+    },
+
+    _onIronResize: function(){
```

Comment:
Conventionally, all public and protected methods should be documented. We haven't documented everything in the past, but as we add new things we are making sure to document them.